examples/step-61/doc/intro.dox

```diff
@@ -0,0 +1,281 @@
+This program is to solve the Poisson equation 
+by the weak Galerkin finite element method:
+@f{eqnarray*}
+  \nabla \cdot \left( -\mathbf{K} \nabla p \right)
+    = f,
+    \quad \mathbf{x} \in \Omega, \\
+  p =  p_D,\quad \mathbf{x} \in \Gamma^D, \\
+  \mathbf{u} \cdot \mathbf{n} = u_N,
+  \quad \mathbf{x} \in \Gamma^N, 
+@f}
+where $\Omega \subset \mathbb{R}^n (n=2,3)$ is a bounded domain. 
+In the context of the flow of a fluid through a porous medium,
+$p$ is the pressure, $\mathbf{K}$ is a permeability tensor, 
+$ f $ is the source term, and
+$ p_D, u_N $ are respectively Dirichlet and Neumann boundary condition
+We can introduce a flux, $\mathbf{u} = -\mathbf{K} \nabla p$, that corresponds 
+to the Darcy velocity (in the way we did in step-20) and this variable will
+be important in the considerations below.
+
+In this program, we will consider a test case where the exact pressure 
+is $p = \sin \left( \pi x)\sin(\pi y \right)$ on the unit square domain,
+with homogenous Dirichelet boundary conditions and identity matrix $\mathbf{K}$.
+Then we will calculate $L_2$ errors of pressure, velocity and flux.
+
+
+<h3> Weak Galerkin scheme </h3>
+
+Via integration by parts, the weak Galerkin scheme for the Poisson equation is
+@f{equation*}
+\mathcal{A}_h\left(p_h,q \right) = \mathcal{F} \left(q \right),
+@f}
+where
+@f{equation*}
+\mathcal{A}_h\left(p_h,q\right)
+  := \sum_{E \in \mathcal{E}_h}
+    \int_E \mathbf{K} \nabla_{w,d} p_h \cdot \nabla_{w,d} q,
+@f}
+and 
+@f{equation*}
+\mathcal{F}\left(q\right)
+  := \sum_{E \in \mathcal{E}_h} \int_E f \, q^\circ
+  - \sum_{\gamma \in \Gamma_h^N} \int_\gamma u_N q^\partial,
+@f}
+$q^\circ$ is the shape function of the polynomial space in the interior, 
+$q^\partial$ is the shape function of the polynomial space on faces. 
+We use FE_DGQ as the interior polynomial space, 
+FE_FaceQ as the face polynomial space. 
+For the numerical velocity $\mathbf{u} = -{\mathbf{K}} \nabla_{w,d} p$, 
+it is defined on the vector space, Raviart-Thomas spaces $RT(k)$.
+
+<h3> Assembling the linear system </h3>
+
+First, we solve for the pressure. 
+We collect two local spaces together in one FESystem, 
+the first component in this finite element system denotes 
+the space for interior pressure, and the second denotes 
+the space for face pressure. 
+For the interior component, we use the polynomial space FE_DGQ. 
+For the face component, we use FE_FaceQ.
+
+We use shape functions defined on spaces FE_DGQ and FE_FaceQ to 
+approximate pressures, i.e., $p_h = \sum a_i \phi_i,$ 
+where $\phi_i$ are shape functions of FESystem. 
+We construct the local system by using discrete weak gradients of 
+shape functions of FE_DGQ and FE_FaceQ.
+The discrete weak gradients of shape functions of 
+FESystem $\nabla_{w,d} \phi$ are defined as 
+$\nabla_{w,d} \phi = \sum_{i=1}^m c_i \mathbf{w}_i,$ 
+where $\mathbf{w}_i$ is the basis function of $RT(k)$, 
+$m$ is the dof of the $RT(k)$. 
```

Comment:
`m` is the number of degrees of freedom of 'RT(k)`?

Reply:
It is the number of degrees of freedom of each Raviart-Thomas element.